### ISSUE TITLE
Ignored methods for Layout/MultilineMethodCallIndentation

### ISSUE THREAD

**Issue by mockdeep** [rubocop/rubocop#9710]
## Is your feature request related to a problem? Please describe.

I like to write RSpec tests like:

```rb
expect { described_class.call(user) }
  .to change(target1, :value).to(0)
  .and change(target2, :value).to(0)
```

But the `Layout/MultilineMethodCallIndentation` cop would have me write them like:

```rb
expect { described_class.call(user) }
  .to change(target1, :value)
  .to(0)
  .and change(target2, :value)
  .to(0)
```

## Describe the solution you'd like

In general, I like the rule, but in these specific scenarios it would be nice if there were exceptions for cases like this. Maybe there are better ideas, but my immediate thought is to be able to allow exceptions for particular method names, like `#to` and `#from`.

## Describe alternatives you've considered

I could stop using the rule, or stop using the `change` matcher, but I don't love either of those. For now I'll probably disable the rule in my tests.

**Comment by bbatsov** [comment]
Hmm, that's weird, as it seems to me this should allowed even today. I assume the omitted parens might be causing some confusing for the cop in question. Do you get the same results if you add the missing parens?

**Comment by mockdeep** [comment]
@bbatsov ah, yeah, good question. If I reformat the code to this RuboCop doesn't complain:

```rb
expect { described_class.call(user) }
  .to(change(target1, :value).to(0)
  .and(change(target2, :value).to(0)))
```

I'm actually a little surprised it allows that, since the `.and` line is part of an argument to the `.to` line.

**Comment by dvandersluis** [comment]
If you use this config:

```yaml
Layout/MultilineMethodCallIndentation:
  EnforcedStyle: indented
```

Your original code will not be flagged by RuboCop.

As well, you could change it like this instead of nesting the calls:

```ruby
expect { described_class.call(user) }
  .to(change(target1, :value).to(0))
  .and(change(target2, :value).to(0))
```

**Comment by paul** [comment]
We're using:

```yaml
Layout/MultilineMethodCallIndentation:
  EnforcedStyle: indented_relative_to_receiver
```

and its wanting to reformat like this:

```ruby
expect { perform }
  .to  change { rel1.reload.exists? }.from(false).to(true)
  .and change { rel2.reload.exists? }.from(false).to(true)
# -- autocorrects to: --
expect { perform }
  .to change { rel1.reload.exists? }.from(false).to(true)
        .and change { rel2.reload.exists? }.from(false).to(true)
```

It indents the `.and` 6 spaces (3 indents), I imagine because of the two methods chained (`.from(...).to(...)`)?

I don't want to change our preferred style is "indented_relative_to_receiver" because we want this style everywhere: 

```ruby
foo = foobar_query
        .baz(123)
        .qux("456" => "789")
```

However, it seems to get confused by the lack of spaces on `.to` and `.and` in these rspec `change` expectations. Maybe it can be modified to not increment the indent counter for methods chained in the same line? (I'm completely speculating on how the cop is implemented.)

**Comment by alexpapworth** [comment]
Seeing this same problem with `indented_relative_to_receiver` as well. Would be nice to see a fix for it.
